.github/scripts/test_covering_pr.js

```diff
@@ -1,30 +1,37 @@
 const Redis = require('ioredis');
-const fs = require('fs').promises;
 
 const main = async () => {
-    const changedFiles = process.argv.slice(2)
+    const changedFiles = process.argv.slice(2);
     const redis = new Redis({
         host: process.env.REDIS_HOST,
         port: process.env.REDIS_PORT,
         username: process.env.REDIS_USER,
         password: process.env.REDIS_PASS,
         enableReadyCheck: false
     });
-    let tests = new Set();
 
     /**
      * Loop over all the files changed in the PR and check if it's covered by a test
      **/
-    for (const filepath of changedFiles) {
-        var classpath = filepath.replace('java/code/src/', '');
-        await redis.smembers(classpath, function(err, test_names) {
-            test_names.forEach(function(test) {
-                tests.add(test);
-            });
-        });
+    try {
+        const tests = new Set();
+
+        for (const filepath of changedFiles) {
+            const classpath = filepath.replace('java/code/src/', '');
+            const testNames = await redis.smembers(classpath);
+            testNames.forEach(test => tests.add(test));
+        }
+
+        console.log(Array.from(tests).join(','));
+    } catch (error) {
+        console.error('Error:', error);
+        process.exit(1);
```

Comment:
Would recommend the following change, that way you'll allow the process to exit gracefully, but with the same error code.

```suggestion
        process.exitCode = 1;
```

Reply:
Thank you! I will apply it here, and below.